components/RoomNav/index.js

```diff
@@ -0,0 +1,78 @@
+import TalkList from "../TalkList";
+import TalkItem from "../TalkItem";
+import TalkInfoWrapper from "../TalkInfoWrapper";
+import FooterNav from "../FooterNav";
+import StyledLink from "../StyledLink";
+import StyledRoom from "../StyledRoom";
+import { useState } from "react";
+import styled from "styled-components";
+
+export default function RoomNav({ talks }) {
+  const rooms = ["Great Hall", "Liberty Hall", "Metro West"];
+  const [roomIndex, setRoomIndex] = useState(0);
+
+  const talksInCurrentRoom = talks
+    .filter((talk) => talk.room === rooms[roomIndex])
+    .sort((a, b) => {
+      let timeA = a.time;
+      let timeB = b.time;
+
+      return timeA < timeB ? -1 : 1;
+    })
+    .sort((a, b) => {
+      let sessionA = a.session;
+      let sessionB = b.session;
+
+      return sessionA < sessionB ? -1 : 1;
+    });
+
+  function handleNextRoomButton() {
+    if (roomIndex < rooms.length - 1) {
+      setRoomIndex((prev) => prev + 1);
+    }
+    if (roomIndex === 2) {
+      setRoomIndex(0);
+    }
+  }
```

Comment:
I think, you do not need to hardcore the number "2". 🤔 It should probably work by setting the `roomIndex` to "0", if the `roomIndex` is equal to `rooms.length-1` (which is "2"), otherwise you can just increase the `roomIndex` by 1.

```suggestion
if (roomIndex === rooms.length-1) {
  roomIndex = 0
} else {
  roomIndex = roomIndex +1;
}
```

Reply:
I tried it but it doesn't work. Jessica and I had a hard time yesterday figuring out a way. Maybe we can try and take a look at it another time together 🤔

Reply:
I updated my comment. Please try it out once more. I forgot to use `setRoomIndex` as I tried that version in a little CodePen before, where I didn't need that 🙈